practices/Ramon-OC/practica01/practica01.py

```diff
@@ -0,0 +1,332 @@
+
+    Returns
+    -------
+    list[str]:
+        List with posible transcriptions if any,
+        else a list with the string "NOT FOUND"
+    """
+    return dataset.get(word.lower(), "NOT FOUND").split(", ")
+
+
+# + id="65a1e3df"
+def word_IPA(searchWord: str, dataset: dict) -> str:
+    """Search for a word in an IPA phonetics dict
+
+      Given a string this function returns IPA representation
+      Parameters:
+      -----------
+      searchWord: str
+          A word to search in the dataset
+      dataset: dict
+          A dataset for a given language code
+      Returns
+      -------
+      str:
+          String with original word.
+      """
+    result = query_ipa_transcriptions(searchWord, dataset)
+    return searchWord + " | " + result[0]
+
+
+# + [markdown] id="0cd27423-8154-48e2-92c2-b527dcc4833a"
+# ### Obteniendo corpora desde GitHub
+
+# + id="YSRb9cx5jM8d"
+lang_codes = {
+  "ar": "Arabic (Modern Standard)",
+  "de": "German",
+  "en_UK": "English (Received Pronunciation)",
+  "en_US": "English (General American)",
+  "eo": "Esperanto",
+  "es_ES": "Spanish (Spain)",
+  "es_MX": "Spanish (Mexico)",
+  "fa": "Persian",
+  "fi": "Finnish",
+  "fr_FR": "French (France)",
+  "fr_QC": "French (Québec)",
+  "is": "Icelandic",
+  "ja": "Japanese",
+  "jam": "Jamaican Creole",
+  "km": "Khmer",
+  "ko": "Korean",
+  "ma": "Malay (Malaysian and Indonesian)",
+  "nb": "Norwegian Bokmål",
+  "nl": "Dutch",
+  "or": "Odia",
+  "ro": "Romanian",
+  "sv": "Swedish",
+  "sw": "Swahili",
+  "tts": "Isan",
+  "vi_C": "Vietnamese (Central)",
+  "vi_N": "Vietnamese (Northern)",
+  "vi_S": "Vietnamese (Southern)",
+  "yue": "Cantonese",
+  "zh": "Mandarin"
+}
+iso_lang_codes = list(lang_codes.keys())
+
+
+# + id="WcCmgrgnT9wK" colab={"base_uri": "https://localhost:8080/"} outputId="5f5e11fa-5a30-4182-d21e-e8411af4f3ba"
+def get_dataset() -> dict:
+    """Download corpora from ipa-dict github
+
+    Given a list of iso lang codes download available datasets.
+
+    Returns
+    -------
+    dict
+        Lang codes as keys and dictionary with words-transcriptions
+        as values
+    """
+    return {code: get_ipa_dict(code) for code in iso_lang_codes}
+
+dataset = get_dataset()
+
+
+# + [markdown] id="bFfv_5FnANgs"
+# 1. Agregar un nuevo modo de búsqueda donde se extienda el comportamiento básico del buscador para ahora buscar por frases. Ejemplo:
+#
+# ```
+# [es_MX]>>> Hola que hace
+#  /ola/ /ke/ /ase/
+# ```
+
+# + id="c445e62f"
+def sentence_IPA(sentence: str, dataset: dict) -> str:
+    """Search for a sentence in an IPA phonetics dict
+
+      Given a sentence this function returns a string with its IPA phonetics
+
+      Parameters:
+      -----------
+      sentence: str
+          A sentence to search in the dataset
+      dataset: dict
+          A dataset for a given language code
+      Returns
+      -------
+      str:
+          String with posible homophones.
+      """
+    x = sentence.split()
```

Comment:
x no es un nombre muy claro, recomiendo usar nombres más descriptivos

```suggestion
    sentence_list = sentence.split()
```